From: Dongwan Kim <kdw6485@gmail.com>
Subject: [PATCH 2/4] riched20: fill REOBJECT.cp when an OLE object inserted with REO_CP_SELECTION
Message-Id: <20211124020106.3704-2-kdw6485@gmail.com>
Date: Wed, 24 Nov 2021 11:01:04 +0900
In-Reply-To: <20211124020106.3704-1-kdw6485@gmail.com>
References: <20211124020106.3704-1-kdw6485@gmail.com>

The character position should be filled in REOBJECT
which IRichEditOle::GetObject returns.

Signed-off-by: Dongwan Kim <kdw6485@gmail.com>
---
 dlls/riched20/caret.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/dlls/riched20/caret.c b/dlls/riched20/caret.c
index 11111111111..11111111111 100644
--- a/dlls/riched20/caret.c
+++ b/dlls/riched20/caret.c
@@ -464,6 +464,7 @@ HRESULT editor_insert_oleobj(ME_TextEditor *editor, const REOBJECT *reo)
   ME_Cursor *cursor, cursor_from_ofs;
   ME_Style *style;
   HRESULT hr;
+  LONG cp = 0;
 
   if (editor->lpOleCallback)
   {
@@ -491,12 +492,16 @@ HRESULT editor_insert_oleobj(ME_TextEditor *editor, const REOBJECT *reo)
   prev = run;
   while ((prev = run_prev_all_paras( prev )))
   {
+    cp += prev->len;
     if (prev->reobj)
     {
       reobj_prev = prev->reobj;
+      cp += prev->reobj->obj.cp;
       break;
     }
   }
+  if(run->reobj->obj.cp == REO_CP_SELECTION)
+    run->reobj->obj.cp = cp;
   if (reobj_prev)
     list_add_after(&reobj_prev->entry, &run->reobj->entry);
   else

-- 
2.36.1

